wagtail/contrib/modeladmin/views.py

```diff
@@ -697,6 +730,7 @@ def check_action_permitted(self, user):
 
     @method_decorator(login_required)
     def dispatch(self, request, *args, **kwargs):
+        self.instance
```

Comment:
I also moved that line into the if statement below, as we only need to worry about prefetching the instance at that time when `self.is_pagemodel`